subprojects/kotlin-dsl-plugins/src/integTest/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPluginCrossVersionSmokeTest.kt

```diff
@@ -71,4 +66,108 @@ class KotlinDslPluginCrossVersionSmokeTest : AbstractKotlinIntegrationTest() {
             )
         }
     }
+
+    @Test
+    fun `can build plugin for oldest supported Kotlin language version`() {
+
+        // Kotlin version leaks on the classpath when running embedded
+        assumeNonEmbeddedGradleExecuter()
+
+        val oldestKotlinLanguageVersion = KotlinGradlePluginVersions.getLANGUAGE_VERSIONS().first()
+
+        withDefaultSettingsIn("producer")
+        withBuildScriptIn("producer", scriptWithKotlinDslPlugin()).appendText(
+            """
+            tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+                compilerOptions {
+                    languageVersion = org.jetbrains.kotlin.gradle.dsl.KotlinVersion.fromVersion("$oldestKotlinLanguageVersion")
+                    apiVersion = org.jetbrains.kotlin.gradle.dsl.KotlinVersion.fromVersion("$oldestKotlinLanguageVersion")
+                }
+            }
+            """
+        )
+        withFile("producer/src/main/kotlin/some.gradle.kts", """println("some!")""")
+
+        withDefaultSettings().appendText("""includeBuild("producer")""")
+        withBuildScript("""plugins { id("some") }""")
+
+        executer.expectDeprecationWarning("w: Language version $oldestKotlinLanguageVersion is deprecated and its support will be removed in a future version of Kotlin")
+
+        build("help").apply {
+            assertThat(output, containsString("some!"))
+        }
+    }
+
+    @Test
+    fun `can build plugin for previous unsupported Kotlin language version`() {
+
+        // Kotlin version leaks on the classpath when running embedded
+        assumeNonEmbeddedGradleExecuter()
+
+        val previousKotlinLanguageVersion = "1.2"
```

Comment:
🤔 Does it make sense to use a Kotlin version that Gradle never really used? Should we stick with 1.3 instead?

Reply:
I captured the oldest we can use. I think it has value given before Gradle 5.0 we already had an older embedded Kotlin, it's just not advertised on the compatibility matrix.